prov/efa/src/efa_domain.c

```diff
@@ -539,6 +538,58 @@ static int efa_set_domain_hmem_info_cuda(struct efa_domain *domain)
 	return 0;
 }
 
+static int efa_set_domain_hmem_info_neuron(struct efa_domain *domain)
+{
+#if HAVE_NEURON
+	struct ibv_mr *ibv_mr;
+	int ibv_access = IBV_ACCESS_LOCAL_WRITE;
+	void *handle;
+	void *ptr = NULL;
+	size_t len = ofi_get_page_size() * 2;
+	int ret;
+
+	if (!ofi_hmem_is_initialized(FI_HMEM_NEURON)) {
+		EFA_INFO(FI_LOG_DOMAIN,
+		         "FI_HMEM_NEURON is not initialized\n");
+		return 0;
+	}
+
+	if (domain->ctx->device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ) {
+		ibv_access |= IBV_ACCESS_REMOTE_READ;
+	} else {
+		EFA_WARN(FI_LOG_DOMAIN,
+			 "No EFA RDMA read support, transfers using AWS Neuron will fail.\n");
+		return 0;
+	}
+
+	domain->hmem_info[FI_HMEM_NEURON].initialized = true;
+
+	ptr = neuron_alloc(&handle, len);
```

Comment:
Where is neuron_alloc and neuron_free defined?

Reply:
They are defined as general functions in the hmem implementation/core, see

https://github.com/ofiwg/libfabric/pull/7339/files#diff-d9867b19501365a647eccf22185bb07763afbfaf1104e0534a5ed4bd10d3ed1dR131